@livarcocc @dsplaisted @rainersigwald Have we seen this before on Ubuntu 14.04:

error MSB4166: Child node "2" exited prematurely. Shutting down. Diagnostic information may be found in files in "/tmp/" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.

It has failed twice in a row. I am going to set MSBUILDDEBUGPATH so that we get those failures in build artifacts.

--- a/build/build.ps1
+++ b/build/build.ps1
@@ -92,6 +92,9 @@ function InitializeDotNetCli {
     $env:DOTNET_INSTALL_DIR = $env:DotNetCoreSdkDir    
   }
 
+  # Save MSBuild crash files info to log directory so that they are captured as build artifacts
+  $env:MSBUILDDEBUGPATH=$LogDir
+
   # Use dotnet installation specified in DOTNET_INSTALL_DIR if it contains the required SDK version, 
   # otherwise install the dotnet CLI and SDK to repo local .dotnet directory to avoid potential permission issues.
   if (($env:DOTNET_INSTALL_DIR -ne $null) -and (Test-Path(Join-Path $env:DOTNET_INSTALL_DIR "sdk\$($GlobalJson.sdk.version)"))) {

--- a/build/build.sh
+++ b/build/build.sh
@@ -167,6 +167,9 @@ function InitializeDotNetCli {
     export DOTNET_INSTALL_DIR="$DotNetCoreSdkDir"
   fi
 
+  # Save MSBuild crash files info to log directory so that they are captured as build artifacts
+  export MSBUILDDEBUGPATH="$log_dir"
+
   ReadJson "$global_json_file" "version"
   local dotnet_sdk_version="$readjsonvalue"
   local dotnet_root=""

